crates/doc/src/extractor.rs

```diff
@@ -0,0 +1,334 @@
+use crate::{compare::compare, ArchivedNode, AsNode, LazyNode, Node, Pointer};
```

Comment:
Why is this in `doc` and not the new `extractors` crate?

Reply:
Other parts of the `doc` crate depend on `doc::Extractor` -- it definitely must live here.

But, `doc` _doens't_ depend on `proto-flow` and I didn't wish to add that (heavy) dependency. Thus the `extractors` crate, which exists merely to bring together `proto-flow` and `doc` and introduce common means of building `doc::Extractors` from `proto_flow::flow::Projection`'s.

Reply:
Huh, okay makes sense!